app/controllers/openid_application_controller.rb

```diff
@@ -94,27 +92,47 @@ def analyze_redis_launch_context
 
   def handle_opaque_token(token_string, aud)
     opaque_token = OpaqueToken.new(token_string, aud)
-    opaque_token.set_payload(fetch_issued(token_string))
-
-    return opaque_token if TokenUtil.validate_token(opaque_token)
-
+    @session = Session.find(hash_token(opaque_token))
+    if @session.nil?
+      opaque_token.set_payload(fetch_issued(token_string))
+      opaque_token.set_ssoi(!opaque_token.static?)
+      return opaque_token if TokenUtil.validate_token(opaque_token)
+    elsif opaque_token.set_ssoi(true)
+      return opaque_token
+    end
     raise error_klass('Invalid token.')
   end
 
-  def populate_ssoi_token_payload(profile)
+  def populate_ssoi_saml_token_payload(profile)
+    token.payload['last_login_type'] = 'ssoi'
+    token.payload['icn'] = profile.attrs['icn']
+    token.payload['npi'] = profile.attrs['npi']
+    token.payload['sec_id'] = profile.attrs['SecID']
+    token.payload['vista_id'] = profile.attrs['VistaId']
+  end
+
+  def populate_ssoi_oauth_token_payload(profile)
     token.payload['last_login_type'] = 'ssoi'
     token.payload['icn'] = profile.attrs['icn']
     token.payload['npi'] = profile.attrs['npi']
     token.payload['sec_id'] = profile.attrs['SecID']
     token.payload['vista_id'] = profile.attrs['VistaId']
+    token.payload['iat'] = profile.attrs['iat']
+    token.payload['exp'] = profile.attrs['exp']
+    token.payload['jti'] = profile.attrs['SecID']
```

Comment:
We probably want to pick different values for `id` and `jti`. These should represent something about the token itself vs the actor.

Reply:
I've updated to use the `fediamtransactionId` as it is unique to the token itself like you mentioned.